### PR TITLE
Add com.github.treagod.spectator

### PR DESCRIPTION
**Pull request by treagod**

### CODE REVIEW

File: com.github.treagod.spectator.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{
+	"app-id": "com.github.treagod.spectator",
+	"base": "io.elementary.BaseApp",
+	"base-version": "juno",
+	"runtime": "org.gnome.Platform",
+	"sdk": "org.gnome.Sdk",
+	"runtime-version": "3.32",
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:7]:
```suggestion
	"runtime-version": "3.36",
```

File: com.github.treagod.spectator.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    "command": "com.github.treagod.spectator",
+    "tags": ["devel", "development" ],
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:9]:
```suggestion
```

File: com.github.treagod.spectator.json [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		"--socket=wayland",
+		"--socket=x11",
+		"--own-name=com.github.treagod.spectator",
+		"--talk-name=org.gtk.vfs",
+		"--talk-name=org.gtk.vfs.*",
+		"--talk-name=org.gnome.SettingsDaemon",
+		"--filesystem=xdg-run/dconf",
+		"--filesystem=xdg-config/dconf:ro",
+		"--talk-name=ca.desrt.dconf",
+		"--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:22]:
```suggestion
```

File: com.github.treagod.spectator.json [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	"sdk": "org.gnome.Sdk",
+	"runtime-version": "3.32",
+    "command": "com.github.treagod.spectator",
+    "tags": ["devel", "development" ],
+	"rename-icon": "com.github.treagod.spectator",
+	"finish-args": [
+		"--filesystem=home",
+		"--socket=wayland",
+		"--socket=x11",
+		"--own-name=com.github.treagod.spectator",
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:15]:
```suggestion
```
you own your app-id by default.

File: com.github.treagod.spectator.json [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+					"type": "archive",
+					"url": "https://github.com/treagod/spectator/archive/0.4.1.zip",
+					"sha256": "c8a37d17126fed5ef68de1eaecc76cef7026c94d1752208808460d976e9303d0"
+				}
+			]
+		}
+	],
+	"build-options": {
+		"env": {}
+	}
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:77]:
```suggestion
```

**Review by treagod — COMMENTED** [review]
Thanks for your feedback!

File: com.github.treagod.spectator.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
+{
+	"app-id": "com.github.treagod.spectator",
+	"base": "io.elementary.BaseApp",
+	"base-version": "juno",
+	"runtime": "org.gnome.Platform",
+	"sdk": "org.gnome.Sdk",
+	"runtime-version": "3.36",
+    "command": "com.github.treagod.spectator",
+	"rename-icon": "com.github.treagod.spectator",
```
Comment (barthalion) [inline com.github.treagod.spectator.json:9]: It does not seem necessary, it's already a correct icon name.

**Comment by barthalion** [comment]
bot, build com.github.treagod.spectator

**Comment by barthalion** [comment]
bot, build com.github.treagod.spectator

**Comment by barthalion** [comment]
bot, build com.github.treagod.spectator

File: com.github.treagod.spectator.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
+{
+  "app-id": "com.github.treagod.spectator",
+  "base": "io.elementary.BaseApp",
+  "base-version": "juno-20.08",
+  "runtime": "org.gnome.Platform",
+  "sdk": "org.gnome.Sdk",
+  "runtime-version": "3.38",
+  "command": "com.github.treagod.spectator",
+  "finish-args": [
+    "--filesystem=home",
```
Comment (barthalion) [inline com.github.treagod.spectator.json:10]: What does it need home access for?

File: com.github.treagod.spectator.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--talk-name=org.gtk.vfs",
+    "--talk-name=org.gtk.vfs.*",
```
Comment (barthalion) [inline com.github.treagod.spectator.json:14]: Same question as above.

File: com.github.treagod.spectator.json (hunk 7 above)
Comment (treagod) [reply com.github.treagod.spectator.json:10]: The application saves the user data in $HOME/.local/share/com.github.treagod.spectator/com.github.treagod.spectator.db

Comment (barthalion) [reply com.github.treagod.spectator.json:10]: You should use `$XDG_DATA_HOME` instead, flatpak grants access there automatically.

Comment (treagod) [reply com.github.treagod.spectator.json:10]: Is it ok if I use `$XDG_DATA_HOME`  in my application if it is defined and `$HOME/.local/share/com.github.treagod.spectator/com.github.treagod.spectator.db` if it is not defined?

Comment (bilelmoussaoui) [reply com.github.treagod.spectator.json:10]: That sounds good, Flatpak ensures it's always defined though but a fallback doesn't hurt.

**Comment by bilelmoussaoui** [comment]
bot, build com.github.treagod.spectator

**Comment by treagod** [comment]
Needed to update the version

**Comment by bilelmoussaoui** [comment]
bot, build com.github.treagod.spectator

File: com.github.treagod.spectator.json [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  "base": "io.elementary.BaseApp",
+  "base-version": "juno-20.08",
+  "runtime": "org.gnome.Platform",
+  "sdk": "org.gnome.Sdk",
+  "runtime-version": "3.38",
+  "command": "com.github.treagod.spectator",
+  "finish-args": [
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--talk-name=org.gnome.SettingsDaemon"
```
Comment (barthalion) [inline com.github.treagod.spectator.json:12]: Is it needed for anything? I took a quick look at the code and it doesn't seem used anywhere.

Comment (barthalion) [reply com.github.treagod.spectator.json:12]: Actually looks like something is bugged although maybe only on my system? Added a GET request to Flathub API but nothing happens after clicking on "play" icon. I see an error when I pass `--log-session-bus` to flatpak:

```
❯ flatpak run --log-session-bus com.github.treagod.spectator
(...)
C33: -> :1.89 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B830: <- :1.89 return from C33
C34: -> :1.89 call org.freedesktop.portal.ProxyResolver.Lookup at /org/freedesktop/portal/desktop
B832: <- :1.89 return error org.freedesktop.portal.Error.NotAllowed from C34
```

Comment (treagod) [reply com.github.treagod.spectator.json]: Actually I can't make any sense of the error message. Is there something I can try to solve the problem?

Comment (barthalion) [reply com.github.treagod.spectator.json]: First of all, does it work for you? Maybe the issue is on my end and everything is fine with the app.

**Comment by nedrichards** [comment]
bot, build com.github.treagod.spectator

**Comment by treagod** [comment]
The application now successfully starts. But making http requests do not work. Is there some configuration to allow outgoing http requests?

File: com.github.treagod.spectator.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
+{
+  "app-id": "com.github.treagod.spectator",
+  "base": "io.elementary.BaseApp",
+  "base-version": "juno-20.08",
+  "runtime": "org.gnome.Platform",
+  "sdk": "org.gnome.Sdk",
+  "runtime-version": "3.38",
+  "command": "com.github.treagod.spectator",
+  "finish-args": [
+    "--socket=wayland",
```
Comment (bilelmoussaoui) [inline com.github.treagod.spectator.json:10]: You need network access here.

Comment (barthalion) [reply com.github.treagod.spectator.json:10]: That's embarrassing, no idea how I missed it. =(

**Comment by barthalion** [comment]
bot, build com.github.treagod.spectator

**Pull request closed without merging**